### PR TITLE
fix(core): fix parallel cli argument not default to 3 when not specified

### LINKED ISSUE #13782: Nx's default `--parallel` value not correctly set to 3
### Current Behavior

If the `--parallel` tag is not explicitly set, then it defaults to `undefined`, although the documentation specifies that 3 is the default value.

### Expected Behavior

It should be defaulted to 3, or the documentation should be updated to say the default value is `undefined`, meaning sequential execution

### Github Repo

https://github.com/nrwl/nx/blob/master/packages/nx/src/utils/command-line-utils.ts#L168-L180

### Steps to Reproduce

1. Run `nx run-many --target=lint --all`, without the `--parallel` tag.
2. All projects are linted sequentially.

### Nx Report

```shell
NA
```

### Failure Logs

```shell
NA
```

### Additional Information

I believe the problem lies within this snippet:
https://github.com/nrwl/nx/blob/master/packages/nx/src/utils/command-line-utils.ts#L168-L180

If `args['parallel']` is `undefined`, which it is when it’s not specified, then `nxArgs['parallel']` is not set and defaults to `undefined` too.

I've tested this by adding this line directly below the snippet above:
```
	output.note({
		title: `Run with args.parallel being '${args['parallel']}', so nxArgs.parallel is set to '${nxArgs['parallel']}'`,
	});
```

Running `nx run-many --target=lint --all` on my repository (or any nx repository for that matter) gives:
```
>  NX   NOTE  Run with args.parallel being 'undefined', so nxArgs.parallel is set to 'undefined'
```

I can create a PR, but I need to know the Nx team's decision. Do you want this `parallel` value to default to 3 or do you want to update the documentation to match the current behaviour?

### PR DESCRIPTION
**Pull request by mike-fam**
## Current Behavior
When the `--parallel` flag is not specified, nx commands run the targets sequentially.

## Expected Behavior
According to the docs, when the `--parallel` flag is not specified, it should default to 3.

## Related Issue(s)
Fixes #13782
BREAKING CHANGE: Default for --parallel is 3

### CODE REVIEW

**Comment by AgentEnder** [comment]
Hey @mike-fam! Talked this one over with @FrozenPandaz this morning. This would be a breaking change for some folks that are expecting sequential targets, so we are going to target this for the next major version.

In the meantime, the docs are attempting to reflect that if passing only `--parallel`, instead of say `--parallel 5`, the default value would be 3. Do you want to open a PR amending that doc to be clearer?

**Comment by mike-fam** [comment]
Hey @AgentEnder, I just want to confirm that having parallel=3 by default when no `--parallel` flag is specified is the behaviour that we want, albeit breaking, right?

**Comment by AgentEnder** [comment]
I do think it makes sense that way, still need to talk it over with a few folks.

**Comment by mike-fam** [comment]
@AgentEnder Have you got any update on this?

File: packages/nx/src/utils/command-line-utils.ts [hunk 1]
```diff
@@ -170,7 +170,8 @@ export function splitArgsIntoNxArgsAndOverrides(
   } else if (
     args['parallel'] === 'true' ||
     args['parallel'] === true ||
-    args['parallel'] === ''
+    args['parallel'] === '' ||
```
Comment (Yaronglp) [inline packages/nx/src/utils/command-line-utils.ts:173]: What is your opinion about creating an array with all the available options and calling the `include()` function to check if `args['parallel']` is matching one of the array's items?

**what are the benefits in my opinion?**
- Readability (easy to understand what are the available options in one place)
- Maintainability (you can always add more cases by adding a new item to the array)

**Comment by AgentEnder** [comment]
Hey @mike-fam, sorry for the delay. I don't have an update on the post v16 target for this, **_for now_**, updating the docs is the best bet.

**Review by AgentEnder — APPROVED** [review]

**Pull request merged**